app/scripts/lib/transaction/util.ts

```diff
@@ -63,10 +63,16 @@ export type AddDappTransactionRequest = BaseAddTransactionRequest & {
 export async function addDappTransaction(
   request: AddDappTransactionRequest,
 ): Promise<string> {
-  const { dappRequest } = request;
+  const { dappRequest, transactionParams } = request;
   const { id: actionId, method, origin } = dappRequest;
   const { securityAlertResponse, traceContext } = dappRequest;
 
+  // Temporary fix for E2E tests that rely on `gasLimit` being ignored
```

Comment:
See here for details on this temporary fix: https://github.com/MetaMask/metamask-extension/pull/30967/files#r1994547176

It was removed shortly after that PR, in https://github.com/MetaMask/metamask-extension/pull/30931/, which was included in v12.16.0. Therefore I don't think we need to bring it into v12.16.0